### PR TITLE
Separating frontend and backend build to run parallel. 

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

## Proposed changes

- Speeds up CircleCI build by building backend and frontend images parallelly
-  previously took 20-23 mins now takes 15-17 mins

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -170,22 +194,36 @@ workflows:
...
+      - setup-env:
           requires:
             - frontend-services-checks
             - backend-services-checks
           filters:
             tags:
               only: /.*/
+      - docker-build-backend:
+          requires:
+            - setup-env
```
Comment (rajdas98) [inline .circleci/config.yml:206]: This is good. Can make it dependent to frontend-services-checks and backend-services-checks as well
```suggestion
            - setup-env
            - frontend-services-checks
            - backend-services-checks
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -170,22 +194,36 @@ workflows:
...
+          filters:
+            tags:
+              only: /.*/
+      - docker-build-frontend:
+          requires:
+            - setup-env
```
Comment (rajdas98) [inline .circleci/config.yml:212]: https://github.com/litmuschaos/litmus/pull/2036/files#r487068594

**Comment by rajdas98** [comment]
Also, just for testing this PR. Can you add `docker images` in the docker-push step? Remove it before PR merge

File: .circleci/config.yml (hunk 1 above)
Comment (gdsoumya) [reply .circleci/config.yml:206]: It is a chain, first the checks complete only then the `setup-env` works so we don't need another dependency here. You can take a look at the current build to see how things are dependant right now.

File: .circleci/config.yml (hunk 2 above)
Comment (gdsoumya) [reply .circleci/config.yml:212]: Same as the above

File: .circleci/config.yml (hunk 1 above)
Comment (rajdas98) [reply .circleci/config.yml:206]: After you add these, all three of these run parallel, which reduces extra the setup-env time

**Comment by gdsoumya** [comment]
```
#!/bin/bash -eo pipefail
docker images
REPOSITORY                               TAG                 IMAGE ID            CREATED             SIZE
litmuschaos/litmusportal-frontend        ci                  42c9e451dbac        52 seconds ago      56.1MB
litmuschaos/litmusportal-self-deployer   ci                  bc1fc3af32c8        8 minutes ago       52.5MB
litmuschaos/litmusportal-subscriber      ci                  0e58d7f3129b        8 minutes ago       45.6MB
litmuschaos/litmusportal-auth-server     ci                  4b0ed48727f4        9 minutes ago       22.5MB
litmuschaos/litmusportal-server          ci                  67bb63854ad8        10 minutes ago      52.3MB
CircleCI received exit code 0
```
Images in CI - Complete build took 14 mins

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**